react-js/react-part-2.md

```diff
@@ -38,180 +38,98 @@ In introduction to [React](/react.md) we discussed about how React makes website
+- In the above instance of setState, an object has been passed as a parameter.
+- In case where previous state of React is used to update the new state(just like above instance), the correct way is to pass a function as a parameter which itself has previous state parameter and that can be used to update state.
+
+```javascript
+addValue = () => {
+  this.setState((prevState) => {
+    return {
+      value: prevState.value +1
     }
-   ```
-
-   ```javascript
-   <input value = {this.state.value} />
-   ```
-
-5. A function must be declared so that when the input box value changes the state should be updated. Let us define a function `onChangeHandler` which will update the state assigned to onChange event handler of input box.
-
-   ```javascript
-   onChangeHandler = (event) => {
-   this.setState({
-     value: [event.target.value]
-   })
-   }
-   ```
-
-   ```javascript
-   <input value = {this.state.value} onChange={this.onChangeHandler} />
-   ```
-
-6. Another function to add the items in the array when `add` button is clicked. This function will be assigned to onClick event handler of add button.
-
-   ```javascript
-   addItem = () => {
-     this.setState({
-       value: "",
-       items: [...this.state.items,this.state.value]
-       })
-    }
-   ```
-
-   ```
-   <button type="button" onClick={this.addItem} >Add</button>
-   ```
-
-7. Now we will render the array items using `map` function.
-
-   ```javascript
-   <ol>
-     {this.state.items.map((item,index) => (
-       <li key={index}>
-        {item}
-       </li>
-      ))}
-    </ol>
-   ```
-
-8. Finally your `App.js` will look like -
-
-    ```javascript
-    import React from 'react';
-    import './App.css';
-
-    class App extends React.Component {
-     state = {
-       items: [],
-       value:""
-     }
-
-    // Function to handle input value of the text field
-     onChangeHandler = (event) => {
-       this.setState({
-         value: [event.target.value]
-       })
-     }
-
-    // Function to add new values to items array
-     addItem = () => {
-       this.setState({
-         value: "",
-         items: [...this.state.items,this.state.value]
-       })
-     }
-
-     render() {
-       return (
-         <div className="App">
-           <input value={this.state.value} onChange={this.onChangeHandler} />
-           <button type="button" onClick={this.addItem} >Add</button>
-           <ol>
-             {this.state.items.map((item,index) => (
-               <li key={index}>
-                 {item}
-               </li>
-              ))}
-            </ol>
-          </div>
-        );
-      }
-    }
-
-    export default App;
-    ```
+  })
+}
+```
+- For now let us use the object instance only.
+
+6. Similarly we will create a function which will decrement the value by 1.
+
+```javascript
+subtractValue = () => {
+  this.setState({
+	  value: this.state.value + 1
```

Comment:
Should this be `value: this.state.value - 1` ?
Correct me if I'm wrong.

Reply:
No, you are right. Typo mistake.